src/components/AppNavigationItem/AppNavigationItem.vue

```diff
@@ -111,97 +111,101 @@ Just set the `pinned` prop.
 </docs>
 
 <template>
-	<!-- Navigation item, can be either an <li> or a <router-link> depending on the props -->
-	<nav-element v-bind="navElement"
+	<element class="app-navigation-entry-wrapper"
 		:class="{
-			'app-navigation-entry--no-icon': !isIconShown,
 			'app-navigation-entry--opened': opened,
-			'app-navigation-entry--pinned': pinned,
-			'app-navigation-entry--editing' : editingActive,
-			'app-navigation-entry--deleted': undo,
 			'app-navigation-entry--collapsible': collapsible,
-			'active': isActive
-		}"
-		class="app-navigation-entry">
-		<!-- Icon and title -->
-		<a v-if="!undo"
-			class="app-navigation-entry-link"
-			:aria-description="ariaDescription"
-			href="#"
-			@click="onClick">
-
-			<!-- icon if not collapsible -->
-			<!-- never show the icon over the collapsible if mobile -->
-			<div :class="{ 'icon-loading-small': loading, [icon]: icon && isIconShown }"
-				class="app-navigation-entry-icon">
-				<slot v-show="!loading && isIconShown" name="icon" />
+		}">
+		<!-- Navigation item, can be either an <li> or a <router-link> depending on the props -->
+		<nav-element v-bind="navElement"
```

Comment:
This needs an `ul` parent, since it's always a `li`

Reply:
https://github.com/nextcloud/nextcloud-vue/blob/d01fcc7402af9f3f773dd575b61caa6e108be6ab/src/components/AppNavigation/AppNavigation.vue#L65-L68

that navigation item always appears on this slot, so there is an ul around it, right?
Sorry, im not sure, what exactly are you proposing to change..

Reply:
Look at the screenshot of the rendered HTML markup above. The `AppNavigationItem` component does not render a `li` anymore, but now gives `element` as its root element. This is not a valid child of `ul` rendered by `AppNavigation`. So the `AppNavigationItem` has to return a `li` element as its root element.

So, I basically just request valid markup (`li`s always need `ul` parents, and `ul` only have `li`s as children), which the current implementation does not give.